### ISSUE TITLE
Feature request: @Nullable support

### ISSUE THREAD

**Issue by cjp39** [google/FreeBuilder#2]
Motivation
-------------

The hardest null bugs to spot are the ones where a method in a different package can return nulls. If you see foo.getBar(), it's unlikely you'll remember the contract of Foo well enough to spot that bar might be null. By contrast, if you see foo.getBar().orNull(), it's impossible to miss—and Optional gives us that. Immutable data types are very likely to be used in exactly these problem areas, so we made a deliberate decision to postpone null support.

That said, with Java 8, we may yet see a truly effective compile-time elimination of NullPointerExceptions from Java code, at which point the key objection to @Nullable is no longer valid.

If you have strong reasons for requiring @Nullable support sooner, please add a comment to this feature request.

Porting existing null-based data types
--------------------------------------------------

If you're upgrading an existing @Nullable-returning data type to @FreeBuilder, Eclipse's "Rename Method" and "Inline Method" refactoring tools make it a cinch to move over to Optional:
  * Rename getBar() to getNullableBar()
  * Create a new, Optional-returning getBar()
  * Change getNullableBar() to simply call getBar().orNull()
  * Inline all instances of getNullableBar()

If you're upgrading an existing Builder, FreeBuilder generates a null-accepting setNullableBar, so just rename the existing setter method accordingly.

**Comment by fluentfuture** [comment]
Just ran into a case where the suggested Optional fix isn't feasible.

I'm trying to generate a FreeBuilder subclass of an existing domain entity interface, that has a nullable getPotentialVolume() method.

The domain interfaces are annotated with validation annotations such as @NotNull, @Required etc. These annotations are bound to certain validators that return meaningful error messages (not just NPE) to users.

Since the validation framework is tied to nulls, changing the @Nullable method to Optional would require adding Optional support to the framework, which isn't trivial.

In summary, I believe nulls are supported in tools and frameworks more widely than the relatively new @Nullable so there is still value to support nulls directly.

**Comment by cjp39** [comment]
Thanks, Ben, you've convinced me. I don't see the benefit in converting legacy code to Optional when compile-time–checked @Nullable annotations are within grasp anyway.
